api/modules/paymentManagement.js

```diff
@@ -22,29 +22,41 @@ const paymentManagement = module.exports = (() => {
     }
 
     const processBitcoinInvoiceCreation = async (paymentId) => {
+        let paymentCurrency
+        let isClientCurrencyBtc = false
+        
         const payment = await db.models.Payment.findByPk(paymentId)
         const { 
             amount, 
             client_id, 
+            currency,
             date_paid, 
             external_uuid, 
             external_uuid_type 
         } = payment.dataValues
 
         if (date_paid) throw new Error(`Payment has already been made`)
 
-        const client = await db.models.Client.findByPk(client_id)
-        const isClientCurrencyBtc = 
-            client.dataValues.currency === `BTC` ||
-            client.dataValues.currency === `SATS`
-        
+        if (client_id) {
+            const client = await db.models.Client.findByPk(client_id)
+            paymentCurrency = client.dataValues.currency
+            isClientCurrencyBtc = 
+            paymentCurrency === `BTC` ||
+            paymentCurrency === `SATS`
+        }
+
+        if (currency) {
+            isClientCurrencyBtc = currency === `BTC` || currency === `SATS`
+            paymentCurrency = currency
+        }
```

Comment:
@rishkwal if the currency is not nullable this will always be true right? Instead of `if` maybe `else` makes more sense so we don't overwrite the other logic

Reply:
You're correct, this is always going to be true.
We can either make `currency` nullable
OR
Maybe we can remove the currency of `client` from the logic and derive the currency only from the Payment
```suggestion
            isClientCurrencyBtc = currency === `BTC` || currency === `SATS`
            paymentCurrency = currency
```
What do you suggest?

Reply:
I would go with this approach to make it more flexible. Even we can default to the project currency if none of those cases are met
```
 if (currency) {
            isClientCurrencyBtc = currency === `BTC` || currency === `SATS`
            paymentCurrency = currency
        }
else if (client_id) {
            const client = await db.models.Client.findByPk(client_id)
            paymentCurrency = client.dataValues.currency
            isClientCurrencyBtc = 
            paymentCurrency === `BTC` ||
            paymentCurrency === `SATS`
        }

    
```